source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/wrappers/torchrl/torchrl_env_wrapper.py

```diff
@@ -0,0 +1,601 @@
+
+        """
+        if self._simple_done:
+            done = tensordict._get_str("done", default=None)
+            any_done = done.any()
+            if any_done:
+                tensordict._set_str(
+                    "_reset",
+                    done.clone(),
+                    validated=True,
+                    inplace=False,
+                    non_blocking=False,
+                )
+        else:
+            any_done = _terminated_or_truncated(
+                tensordict,
+                full_done_spec=self.output_spec["full_done_spec"],
+                key="_reset",
+            )
+
+        return tensordict
+
+    def step(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Makes a step in the environment.
+
+        Step accepts a single argument, tensordict, which usually carries an 'action' key which indicates the action
+        to be taken.
+        Step will call an out-place private method, _step, which is the method to be re-written by EnvBase subclasses.
+
+        Args:
+            tensordict (TensorDictBase): Tensordict containing the action to be taken.
+                If the input tensordict contains a ``"next"`` entry, the values contained in it
+                will prevail over the newly computed values. This gives a mechanism
+                to override the underlying computations.
+
+        Returns:
+            the input tensordict, modified in place with the resulting observations, done state and reward
+            (+ others if needed).
+
+        """
+        # sanity check
+        self._assert_tensordict_shape(tensordict)
+        next_preset = tensordict.get("next", None)  # noqa: SIM910
+
+        next_tensordict = self._step(tensordict)
+        next_tensordict = self._step_proc_data(next_tensordict)
+        if next_preset is not None:
+            # tensordict could already have a "next" key
+            # this could be done more efficiently by not excluding but just passing
+            # the necessary keys
+            next_tensordict.update(next_preset.exclude(*next_tensordict.keys(True, True)))
+        tensordict.set("next", next_tensordict)
+        # Information from the "extras" dict of dicts passed from IsaacLab contains reward and metrics information.
+        # The _data postfix is added to distinguish the field containing the actual values from the dummy {key} field
+        # in the tensordict required for faster rollouts. To get logging to work, we need to move the data from the "next" tensordict
+        # to the top level tensordict.
+        # TODO: Find a more elegant solution to enable rewards and metrics logging. "
+        for key in self.env.unwrapped.extras:
+            tensordict[f"{key}_data"] = tensordict["next"][f"{key}_data"]
+        return tensordict
+
+    def _step(self, tensordict: TensorDictBase) -> TensorDictBase:
+        action = tensordict.get(self.action_key)
```

Comment:
Are actions always only a single tensor? 
This could break if you have multiple action tensors